cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -2842,6 +2845,121 @@ var _ = Describe("Operation Controller", func() {
 				Expect(dummyApplicationSpec.Spec).ToNot(Equal(applicationCR.Spec))
 			})
 		})
+
+		Context("Test getDBOperationForEvent function", func() {
+			var (
+				mockCtrl           *gomock.Controller
+				mockDB             *mocks.MockDatabaseQueries
+				mockClient         *mocks.MockClient
+				mockLogger         *mocks.MockLogSink
+				ctx                context.Context
+				log                logr.Logger
+				operationName      string
+				operationNamespace string
+			)
+
+			BeforeEach(func() {
+				mockCtrl = gomock.NewController(GinkgoT())
+				mockDB = mocks.NewMockDatabaseQueries(mockCtrl)
+				mockClient = mocks.NewMockClient(mockCtrl)
+				mockLogger = mocks.NewMockLogSink(mockCtrl)
+				logger = log.WithSink(mockLogger)
+				ctx = context.Background()
+				operationName = "test-operation"
+				operationNamespace = "test-namespace"
+			})
+
+			AfterEach(func() {
+				mockCtrl.Finish()
+			})
+			It("should successfully get a DB operation", func() {
+				objectKey := types.NamespacedName{Name: operationName, Namespace: operationNamespace}
+
+				mockClient.EXPECT().Get(ctx, objectKey, gomock.Any()).DoAndReturn(func(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+					return nil
+				})
+				mockDB.EXPECT().GetOperationById(ctx, gomock.Any()).DoAndReturn(func(ctx context.Context, dbOp *db.Operation) error {
+					// Set the necessary fields in 'dbOp' to simulate the fetched DB operation
+					dbOp.Instance_id = "test-instance-id"
+					dbOp.Resource_id = "test-resource-id"
+					dbOp.Resource_type = ""
+					return nil
+				})
+
+				newEvent := operationEventLoopEvent{
+					request: newRequest(operationNamespace, operationName),
+					client:  mockClient,
+				}
+				result, err := getDBOperationForEvent(ctx, newEvent, mockDB, logger)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).ToNot(BeNil())
```

Comment:
I think it would be better to verify if result object has expected data in fields.